source/secure_tunneling.c

```diff
@@ -441,14 +437,17 @@ static int s_aws_secure_tunnel_set_connection_id(
             };
 
             s_aws_secure_tunnel_remove_connection_id(secure_tunnel, &reset_message);
+
             if (secure_tunnel->config->on_connection_reset) {
                 secure_tunnel->config->on_connection_reset(
                     &reset_message,
                     AWS_ERROR_IOTDEVICE_SECURE_TUNNELING_INVALID_CONNECTION_ID,
                     secure_tunnel->config->user_data);
             }
 
-            aws_secure_tunnel_connection_reset(secure_tunnel, &reset_message);
+            if (aws_secure_tunnel_connection_reset(secure_tunnel, &reset_message)) {
+                return AWS_OP_ERR;
```

Comment:
The `aws_last_error` should collect the error that was raised during `aws_secure_tunnel_connection_reset`. Tracing back through why that function could/would fail, it'd probably end up due to a validation error.